### PR TITLE
Added documentation

### PR DESCRIPTION
**Pull request by Puneetha17**
Signed-off-by: Puneetha <puneetha17@gmail.com>

### CODE REVIEW

File: protofiles/messages.proto [hunk 1]
```diff
@@ -3,43 +3,79 @@ package protofiles;
 
 import "google/api/annotations.proto";
 
+option csharp_namespace = "Blk-io.Protobuf.Messages";
+option go_package = "github.com/blk-io/chimera-api";
+option java_package = "com.blk-io.chimera-api";
```
Comment (conor10) [inline protofiles/messages.proto]: this should be `io.blk.chimera.api`

File: protofiles/messages.proto [hunk 2]
```diff
@@ -3,43 +3,79 @@ package protofiles;
 
 import "google/api/annotations.proto";
 
+option csharp_namespace = "Blk-io.Protobuf.Messages";
```
Comment (conor10) [inline protofiles/messages.proto]: This should be `Blkio.Chimera.Api`

File: protofiles/messages.proto [hunk 3]
```diff
@@ -3,43 +3,79 @@ package protofiles;
...
+option go_package = "github.com/blk-io/chimera-api";
+option java_package = "com.blk-io.chimera-api";
+option java_outer_classname = "Protofiles";
+option java_multiple_files = true;
+option objc_class_prefix = "BPB";
+
+// `messages` contains serialized protocol buffer messages used to communicate with Crux nodes
```
Comment (conor10) [inline protofiles/messages.proto]: Please can we say Chimera nodes instead of Crux?

File: protofiles/messages.proto [hunk 4]
```diff
@@ -3,43 +3,79 @@ package protofiles;
...
+// The gRPC service definitions using the below proto files are defined in
+// `github.com/blk-io/chimera-api/grpc.proto` -- Refer to the same for example usage
+
+// The Apiversion on which the server is running
 message ApiVersion{
     string version = 1;
 }
+
+// The Upcheck response from the server when used in `upcheck` is called
```
Comment (conor10) [inline protofiles/messages.proto]: Please say: The server response from an `upcheck` request

File: protofiles/messages.proto [hunk 5]
```diff
@@ -3,43 +3,79 @@ package protofiles;
...
     string message = 1;
 }
+
+// The request to be sent to server to add a transaction
 message SendRequest{
+    // The payload of the transaction to be sent
     bytes payload = 1;
+    // The key of the Sender
     string from = 2;
+    // The keys of all the Receipients
```
Comment (conor10) [inline protofiles/messages.proto]: The keys of all the Receipients who are privy to the transaction

File: protofiles/messages.proto [hunk 6]
```diff
@@ -3,43 +3,79 @@ package protofiles;
...
     repeated string to = 3;
 }
+// Response from the server for the SendRquest
```
Comment (conor10) [inline protofiles/messages.proto]: Typo: SendRquest instead of SendRequest

File: protofiles/messages.proto [hunk 7]
```diff
@@ -3,43 +3,79 @@ package protofiles;
...
 message SendResponse{
+    // The transaction hash is returned
     bytes key = 1;
 }
+
+// The Request Query to the server to Receive transaction
```
Comment (conor10) [inline protofiles/messages.proto]: Retrieve a transaction payload

File: protofiles/messages.proto [hunk 8]
```diff
@@ -3,43 +3,79 @@ package protofiles;
...
 message ApiVersion{
     string version = 1;
 }
+
+// The Upcheck response from the server when used in `upcheck` is called
 message UpCheckResponse{
     string message = 1;
 }
+
+// The request to be sent to server to add a transaction
```
Comment (conor10) [inline protofiles/messages.proto]: Store a transaction payload

File: protofiles/messages.proto [hunk 9]
```diff
@@ -3,43 +3,79 @@ package protofiles;
...
     bytes key = 1;
+    // The Senders key
     string to = 2;
 }
+// Response for ReceiveRequest -> contains the transaction payload
 message ReceiveResponse{
     bytes payload = 1;
 }
+
+// The information regarding the node
```
Comment (conor10) [inline protofiles/messages.proto]: Node information

File: protofiles/messages.proto [hunk 10]
```diff
@@ -3,43 +3,79 @@ package protofiles;
...
+option java_outer_classname = "Protofiles";
+option java_multiple_files = true;
+option objc_class_prefix = "BPB";
+
+// `messages` contains serialized protocol buffer messages used to communicate with Crux nodes
+// The gRPC service definitions using the below proto files are defined in
+// `github.com/blk-io/chimera-api/grpc.proto` -- Refer to the same for example usage
+
+// The Apiversion on which the server is running
 message ApiVersion{
```
Comment (conor10) [inline protofiles/messages.proto]: Please can you include a space before the `{`

File: protofiles/messages.proto [hunk 11]
```diff
@@ -3,43 +3,79 @@ package protofiles;
...
     map<string, bytes> recipients = 2;
+    // All the URLs opf the connected nodes
     map<string, bool> parties = 3;
 }
+// Payload consisting of the partyinfo of the connected party
 message PartyInfoResponse{
     bytes payload = 1;
 }
+
+// The Encrypted payload used to Push the new transaction payload to other Crux nodes
```
Comment (conor10) [inline protofiles/messages.proto]: Please say propogate instead of Push.

File: protofiles/grpc.proto [hunk 12]
```diff
@@ -4,33 +4,49 @@ package protofiles;
...
 
+    // Update the partyinfo with response from all the nodes connnected to the node
     rpc UpdatePartyInfo(PartyInfo) returns (PartyInfoResponse) {
         option (google.api.http) = {
             post: "/partyinfo"
             body: "*"
         };
     }
 
+    // By gossip protocol, the Crux node sends the updated transaction to all connected nodes.
```
Comment (conor10) [inline protofiles/grpc.proto]: Propogate payload to remote node

File: protofiles/grpc.proto [hunk 13]
```diff
@@ -4,33 +4,49 @@ package protofiles;
 import "google/api/annotations.proto";
 import "messages.proto";
```
Comment (conor10) [inline protofiles/grpc.proto:6]: should we have the Java, C#, etc options in this file too?

File: protofiles/grpc.proto [hunk 14]
```diff
@@ -4,33 +4,49 @@ package protofiles;
...
-
+    // Checks if the server is running
     rpc Upcheck(UpCheckResponse) returns (UpCheckResponse) {
         option (google.api.http) = {
             post: "/upcheck"
             body: "*"
         };
     }
 
+    // Used to send a transaction request to Crux node
```
Comment (conor10) [inline protofiles/grpc.proto]: Used to store a transaction in the Chimera node

File: protofiles/grpc.proto [hunk 15]
```diff
@@ -4,33 +4,49 @@ package protofiles;
...
     rpc Send(SendRequest) returns (SendResponse);
-
+    // Receive response if any from node.
```
Comment (conor10) [inline protofiles/grpc.proto]: Retrieve a transaction from a Chimera node

File: protofiles/grpc.proto (hunk 13 above)
Comment (Puneetha17) [reply protofiles/grpc.proto:6]: We are importing messages.proto, So I am not sure if it will be redundant.

File: protofiles/grpc.proto [hunk 16]
```diff
@@ -4,33 +4,49 @@ package protofiles;
 import "google/api/annotations.proto";
 import "messages.proto";
 
+// `grpc` contains Simple service definition which can be used to communicate with Crux
```
Comment (conor10) [inline protofiles/grpc.proto]: It still show crux here

File: protofiles/grpc.proto [hunk 17]
```diff
@@ -4,33 +4,49 @@ package protofiles;
...
         option (google.api.http) = {
             post: "/upcheck"
             body: "*"
         };
     }
 
+    // Used to store a transaction in the Chimera node
     rpc Send(SendRequest) returns (SendResponse);
-
+    // Retrieve a transaction from a Chimera node
```
Comment (conor10) [inline protofiles/grpc.proto]: Sorry, I should have said `the Chimera node`

File: protofiles/messages.proto [hunk 18]
```diff
@@ -3,43 +3,79 @@ package protofiles;
 
 import "google/api/annotations.proto";
 
-message ApiVersion{
+option csharp_namespace = "Blkio.Chimera.Api";
+option go_package = "github.com/blk-io/chimera-api";
+option java_package = "io.blk.chimera-api";
```
Comment (conor10) [inline protofiles/messages.proto]: it should be `io.blk.chimera.api`

File: protofiles/messages.proto [hunk 19]
```diff
@@ -3,43 +3,79 @@ package protofiles;
...
+option java_outer_classname = "Protofiles";
```
Comment (conor10) [inline protofiles/messages.proto]: lets get rid of this line

File: protofiles/messages.proto [hunk 20]
```diff
@@ -3,43 +3,79 @@ package protofiles;
...
+option java_multiple_files = true;
+option objc_class_prefix = "BPB";
```
Comment (conor10) [inline protofiles/messages.proto]: Let's get rid of this one too

**Review by conor10 — COMMENTED** [review]
A few more small changes.

File: protofiles/grpc.proto (hunk 13 above)
Comment (conor10) [reply protofiles/grpc.proto:6]: I have a feeling you will need it still

**Pull request merged**